mne/io/reference.py

```diff
@@ -392,7 +392,7 @@ def set_eeg_reference(inst, ref_channels='average', copy=True,
         logger.info('EEG data marked as already having the desired reference.')
     else:
         logger.info('Applying a custom %s '
-                    'reference.' % DEFAULTS['titles'][type_])
+                    'reference.' % DEFAULTS['titles'][ch_type])
```

Comment:
This suggests we have some branches not covered properly. Can you add a loop [here](https://github.com/mne-tools/mne-python/blob/002a5cfc9baecdd0ecdaff1f38555b44a141de35/mne/io/tests/test_reference.py#L236-L238) to do something like this:
```
    for ch_type in ('auto', 'ecog'):
        raw = RawArray(data, create_info(3, 1000., ['ecog'] * 2 + ['misc']))
        reref, ref_data = set_eeg_reference(raw.copy(), ch_type=ch_type)
        _test_reference(raw, reref, ref_data, ['0', '1'])
```
This would fail on `master` but pass on this PR, right?